lib/internal/http2/core.js

```diff
@@ -970,9 +970,8 @@ class Http2Session extends EventEmitter {
 
     const state = stream[kState];
     if (state.rst) {
-      // rst has already been called, do not call again,
-      // skip straight to destroy
-      stream.destroy();
+      // rst has already been called by self or peer,
+      // do not call again
```

Comment:
Also to expand on what this says, we can't throw here because it's possible we received an RST frame from the remote peer and are getting ready to destroy but have not done that yet, so the user code calls `rstStream` in the meantime — that's where it should just gracefully exit rather than throw.

Reply:
Can you `assert(stream.destroyed);` here instead? If I understand correctly, that should always be true?

Reply:
Not necessarily since the destruction gets scheduled using `setImmediate` or `process.nextTick`.